examples/Common patterns for connectors/pagination/marketstack/connector.py

```diff
@@ -0,0 +1,219 @@
+    Args:
+        api_key (String): The api token for accessing data
+        symbols (String): Ticker for which price is to be calculated
+        ticker_start_cursor (String): Starting date to fetch records
+        ticker_end_cursor (String): End date to fetch records
+    Raises:
+        Exception: When request fails or ticker prices cannot be fetched from response
+    Returns:
+        list: ticker prices for particular ticker
+    """
+    ticker_price_offset = 0
+    insert_ticker_price_records = []
+    while True:
+        params = {
+            'access_key': api_key,
+            'symbols': symbols,
+            'limit': 1000,
+            'offset': ticker_price_offset,
+            'date_from': ticker_start_cursor,
+            'date_to': ticker_end_cursor
+        }
+        try:
+            api_result = rq.get(
+                'http://api.marketstack.com/v1/eod', params)
+            api_response = api_result.json()
+            insert_ticker_price_records_temp = api_response["data"]
+            if insert_ticker_price_records_temp:
+                insert_ticker_price_records += insert_ticker_price_records_temp
+                ticker_price_offset += 1000
+            else:
+                break
+        except rq.exceptions.RequestException as e:
+            raise Exception(f"Request failed: {str(e)}")
+        except json.JSONDecodeError as e:
+            raise Exception(f"Failed to parse JSON response: {str(e)}")
+        except KeyError as e:
+            raise Exception(f"Missing expected key in response: {str(e)}")
+        except Exception as e:
+            raise RuntimeError(
+                "Failed Fetching ticker prices, Error: " + str(e))
+
+    return insert_ticker_price_records
+
+
+def initialize_state(state: dict):
+    """This is a function to initialize state
+    Args:
+        state (json): State of the connector
+    Returns:
+        json: State of the connector
+    """
+
+    if not state:
+        state["ticker_offset"] = 0
+        state["ticker_start_cursor"] = "2000-01-01"
+        state["ticker_end_cursor"] = str(date.today())
+
+    # Fetch data till the latest date if ticker_offset is 0
+    if state["ticker_offset"] == 0:
+        state["ticker_end_cursor"] = str(date.today())
+    return state
+
+
+# This creates the connector object that will use the update and schema functions defined in this connector.py file.
+connector = Connector(update=update, schema=schema)
+
+# Check if the script is being run as the main module.
+# This is Python's standard entry method allowing your script to be run directly from the command line or IDE 'run' button.
+# This is useful for debugging while you write your code. Note this method is not called by Fivetran when executing your connector in production.
+# Please test using the "fivetran debug" command prior to finalizing and deploying your connector.
+if __name__ == "__main__":
```

Comment:
Pass configuration.json to debug function so that anyone debugging via IDE would get the confugurations